src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/DuplicatedDependenciesSnapshotFilter.cs

```diff
@@ -30,52 +30,59 @@ internal sealed class DuplicatedDependenciesSnapshotFilter : DependenciesSnapsho
             IAddDependencyContext context)
         {
             IDependency matchingDependency = null;
-            foreach ((string _, IDependency x) in context)
+            bool shouldApplyAlias = false;
+
+            foreach ((string _, IDependency other) in context)
             {
-                if (x.TopLevel 
-                     && !x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
-                     && StringComparers.DependencyProviderTypes.Equals(x.ProviderType, dependency.ProviderType)
-                     && x.Caption.Equals(dependency.Caption, StringComparison.OrdinalIgnoreCase))
+                if (!other.TopLevel ||
+                    other.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase) ||
+                    !StringComparers.DependencyProviderTypes.Equals(other.ProviderType, dependency.ProviderType))
                 {
-                    matchingDependency = x;
-                    break;
+                    continue;
                 }
-            }
 
-            // If found node with same caption, or if there were nodes with same caption but with Alias already applied
-            // NOTE: Performance sensitive, so avoid formatting the Caption with parenthesis if it's possible to avoid it.
-            bool shouldApplyAlias = matchingDependency != null;
-            if (!shouldApplyAlias)
-            {
-                int adjustedLength = dependency.Caption.Length + " (".Length;
-                foreach ((string _, IDependency x) in context)
+                if (other.Caption.StartsWith(dependency.Caption, StringComparison.OrdinalIgnoreCase))
                 {
-                    if (x.TopLevel
-                         && !x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
-                         && StringComparers.DependencyProviderTypes.Equals(x.ProviderType, dependency.ProviderType)
-                         && x.Caption.StartsWith(dependency.Caption, StringComparison.OrdinalIgnoreCase)
-                         && x.Caption.Length >= adjustedLength
-                         && string.Compare(x.Caption, adjustedLength, x.OriginalItemSpec, 0, x.OriginalItemSpec.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                    if (other.Caption.Length == dependency.Caption.Length)
                     {
+                        // Exact match.
+                        System.Diagnostics.Debug.Assert(matchingDependency == null, "matchingDependency should be null");
```

Comment:
Assumes.Null.

Reply:
`Assert` is conditional so wouldn't be in release code.

It's impossible for this value to be anything other than `null`, given the `break`. I'll just remove the assert.